cloudevents/extensions/recordedtime.md

```diff
@@ -0,0 +1,68 @@
+# Recorded Time Extension
+
+This extension defines an attribute that represents the time when an
+[_occurrence_](../spec.md#occurrence)
+was recorded in a particular
+[_event_](../spec.md#event),
+which is the time when the CloudEvent was created by a producer.
+
+This attribute is distinct from the [`time`
+attribute](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#time),
+which, according to the CloudEvents specification, SHOULD be the time when the
+occurrence happened, if it can be determined.
+
+This attribute makes it possible to represent
+[bitemporal](https://en.wikipedia.org/wiki/Bitemporal_modeling) data with
+CloudEvents so that, for every event, both of the following times can be known
+by consumers:
+
+- _Occurrence time_: timestamp of when the occurrence recorded in the event
+  happened, which corresponds to the `time` attribute.
+- _Recorded time_: the timestamp of when the occurrence was recorded in a
+  specific CloudEvent instance, which is represented by this extension.
+
+## Attributes
+
+### recordedtime
+
+- Type: `Timestamp`
+- Description: Timestamp of when the occurrence was recorded in this CloudEvent,
+  i.e. when the CloudEvent was created by a producer.
+- Constraints:
+  - REQUIRED
+  - If present, MUST adhere to the format specified in
```

Comment:
Is there a constraint that this must be "at or later than" the occurrence time? Logically that would make sense, but if there are different clocks involved, I could imagine it not actually being strictly true in all cases.

Reply:
I would say it should be later than the occurrence time, but as you say, it may be that there are different clocks involved, and the system creating CE may only be able to trust its own clock and may not want to assert this constraint, so I would not say it must be.

I will add this to the constraints:

```
- SHOULD be at or later than the _occurrence time_.
```